src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java

```diff
@@ -26,34 +31,53 @@
   private final DatasetRepository datasetRepository;
   private final AsyncDatasetPublishService publishService;
 
+  private final TransformationService transformationService;
+
   public DatasetServiceImpl(
       DatasetGeneratorService generatorService,
       DatasetRepository datasetRepository,
-      AsyncDatasetPublishService publishService) {
+      AsyncDatasetPublishService publishService,
+      TransformationService transformationService) {
     this.generatorService = generatorService;
     this.datasetRepository = datasetRepository;
     this.publishService = publishService;
+    this.transformationService = transformationService;
   }
 
   @Transactional
   @Override
   public Dataset createDataset(String datasetName, Country country, Language language,
-      List<ByteArrayInputStream> records) {
+      List<ByteArrayInputStream> records, InputStream xsltTransformerEDMExternal) {
     requireNonNull(datasetName, "Dataset name must not be null");
     requireNonNull(country, "Country must not be null");
     requireNonNull(language, "Language must not be null");
     requireNonNull(records, "Records must not be null");
     checkArgument(!records.isEmpty(), "Records must not be empty");
 
-    var entity = new DatasetEntity(datasetName, records.size(), language, country);
+    DatasetEntity entity = new DatasetEntity(datasetName, records.size(), language, country);
+
     try {
       entity = datasetRepository.save(entity);
     } catch (Exception e) {
       throw new ServiceException(format("Error creating dataset: [%s]. ", datasetName), e);
     }
 
-    var datasetId = String.valueOf(entity.getDatasetId());
-    var dataset = generatorService
+    String datasetId = String.valueOf(entity.getDatasetId());
+
+    // Extra transformation step occurs here
```

Comment:
This comment whispers "I would like to be a method of my own" :D

Reply:
Ahahah that's true! I will take care of that ;)